src/formatDate.js

```diff
@@ -50,7 +50,31 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const initialSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const normalizedDate = date.split(initialSeparator);
+  const newFormat = toFormat;
+
+  const dateObject = {
+    'DD': 0,
+    'MM': 0,
+    'YY': 0,
+    'YYYY': 0,
+  };
+
+  for (let item = 0; item < 3; item++) {
+    dateObject[fromFormat[item]] = normalizedDate[item];
+  }
+
+  dateObject['YY'] = dateObject['YY'] === 0
+    ? dateObject['YYYY'].slice(-2) : dateObject['YY'];
+
+  const longDate = (dateObject['YY'] >= 30 ? '19' : '20') + dateObject['YY'];
+
+  dateObject['YYYY'] = dateObject['YYYY'] === 0
+    ? longDate : dateObject['YYYY'];
+
```

Comment:
In your decision, you calculate all the possible variants of the year, I'm not sure if this is the best solution. With a lot of work, this can affect performance, but our task is small, so in general it is ok

Reply:
here, i add both 'YY' and 'YYYY' with corresponding values. i find that the easiest way and that allows me to map it to the result, no matter what format the year should be. a ternary operator is a simple if statement. i don't see how it would make a drastic performance hit